pinot-transport/src/main/java/com/linkedin/pinot/routing/builder/DefaultRealtimeRoutingTableBuilder.java

```diff
@@ -0,0 +1,100 @@
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.routing.builder;
+
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.configuration.Configuration;
+import org.apache.helix.model.ExternalView;
+import org.apache.helix.model.InstanceConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.linkedin.pinot.common.response.ServerInstance;
+import com.linkedin.pinot.routing.RoutingTableLookupRequest;
+import com.linkedin.pinot.routing.ServerToSegmentSetMap;
+import com.linkedin.pinot.transport.common.SegmentIdSet;
+
+/**
+ * Create a given number of routing tables based on random selections from ExternalView.
+ */
+public class DefaultRealtimeRoutingTableBuilder extends AbstractRoutingTableBuilder {
+  private static final Logger LOGGER = LoggerFactory.getLogger(DefaultRealtimeRoutingTableBuilder.class);
+  private RoutingTableBuilder _realtimeHLCRoutingTableBuilder;
+  private RoutingTableBuilder _realtimeLLCRoutingTableBuilder;
+
+  @Override
+  public void init(Configuration configuration) {
+    _realtimeHLCRoutingTableBuilder = new KafkaHighLevelConsumerBasedRoutingTableBuilder();
+    _realtimeLLCRoutingTableBuilder = new KafkaLowLevelConsumerRoutingTableBuilder();
+    _realtimeHLCRoutingTableBuilder.init(configuration);
+    _realtimeLLCRoutingTableBuilder.init(configuration);
+  }
+
+  @Override
+  public void computeRoutingTableFromExternalView(String tableName, ExternalView externalView, List<InstanceConfig> instanceConfigList) {
+    _realtimeHLCRoutingTableBuilder.computeRoutingTableFromExternalView(tableName, externalView, instanceConfigList);
+    _realtimeLLCRoutingTableBuilder.computeRoutingTableFromExternalView(tableName, externalView, instanceConfigList);
+  }
+
+  @Override
+  public boolean isEmpty() {
+    return _realtimeHLCRoutingTableBuilder.isEmpty() && _realtimeLLCRoutingTableBuilder.isEmpty();
+  }
+
+  @Override
+  public Map<ServerInstance, SegmentIdSet> findServers(RoutingTableLookupRequest request) {
+    boolean forceLLC = false;
+    boolean forceHLC = false;
+    for (String routingOption : request.getRoutingOptions()) {
+      if (routingOption.equalsIgnoreCase("FORCE_HLC")) {
+        forceHLC = true;
+      }
+
+      if (routingOption.equalsIgnoreCase("FORCE_LLC")) {
+        forceLLC = true;
+      }
+    }
+    if (forceHLC && forceLLC) {
+      throw new RuntimeException("Trying to force routing to both HLC and LLC at the same time");
+    }
+
+    Map<ServerInstance, SegmentIdSet> serverToSegmentSetMaps;
+    if (forceLLC) {
```

Comment:
Agree, I will check with Subbu and remove bunch of features we developed for migration from HLC to LLC